lib/browser.ts

```diff
@@ -672,17 +673,14 @@ export class ProtractorBrowser extends Webdriver {
 
     if (this.ignoreSynchronization) {
       this.driver.get(destination);
-      return this.driver.controlFlow().execute(() => {
-        return this.plugins_.onPageLoad();
-      });
+      return this.driver.controlFlow().execute(() => this.plugins_.onPageLoad()).then(() => {});
```

Comment:
That's a good question. First of all, it doesn't do nothing, it converts `Promise<any[]>` to `Promise<void>`, so that the return type of `ProtractorBrowser#get` is `Promise<void>`.

`onPageLoad` used to be typed simply as `Function`, but now that it's type is inferred to be `(...args: any[]) => webdriver.promise.Promise<any[]>`, it became apparent that the returned promise is resolved with an array (the result of the `.all(...)` call in `pluginFunFactory`), and this array seems to be totally useless. Moreover, the presence of this array can also have unexpected effects in run time. E.g. we may pass a function to `then` expecting that its argument would be undefined whereas in fact it'd get an array.

That's why I'm thinking about moving `.then(() => {})` from here to `Plugins#pluginFunFactory`, so that it creates functions of type `(...args: any[]) => q.Promise<void>`. What are your thoughts on this? Are probably among those functions (`setup`, `onPrepare`, etc.) such whose resolved array value isn't meaningless?